core/src/main/java/org/bitcoinj/core/BitcoinSerializer.java

```diff
@@ -205,7 +205,7 @@ private Message makeMessage(String command, int length, byte[] payloadBytes, byt
         } else if (command.equals("getheaders")) {
             message = new GetHeadersMessage(params, payloadBytes);
         } else if (command.equals("tx")) {
-            message = makeTransaction(payloadBytes, 0, length, hash);
```

Comment:
I think this will be fixed once #1438 is merged. Would you like to review that PR?